Docs review for v2.https

joehan opened the pull request:
### Description
Reviewed the staged HTTPS docs, and found +fixed the following issues:
Formatting Issues: 
- FunctionsErrorCode includes bullet points that were being rendered poorly. Added newlines to fix the formatting
Missing descriptions:
onCall
onRequest
httpError.toJson - Added a brief description
https.HttpsOptions.cors - Added an explanation of what this option does with various types - please double check this to make sure I'm correct
request.rawBody - added a brief description

inlined commented on src/common/providers/https.ts:
```diff
@@ -176,36 +177,52 @@ export interface CallableRequest<T = any> {
  *
  * Possible values:
+ *
  * - 'cancelled': The operation was cancelled (typically by the caller).
```
Should "cancelled" and others be in code font?

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
-  /* HTTP functions can override and specify more than one regions. */
+  /** HTTP functions can override global options and can specify multiple regions to deploy to. */
   region?: SupportedRegion | string | Array<SupportedRegion | string>;
+  /** If true, CORS will be allowed for requests to the function.
```
Let's add a comment that `cors` defaults to true for CallableFunctions.

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
   req: Request,
   res: express.Response
 ) => void | Promise<void>) & {
   __trigger?: unknown;
```
@alpha to remove docs for this and __endpoint

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+ * A callable method for clients to call using a Firebase SDK.
+ */
 export interface CallableFunction<T, Return> extends HttpsFunction {
+  /**@hidden */
```
We document run as a method used for testing.

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
   run(data: CallableRequest<T>): Return;
 }
+/**
+ * Handle HTTPS requests
```
Handles (Imagine the messages stars with "this method")

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
 }
+/**
+ * Handle HTTPS requests
+ * @param opts Options to set on this Cloud Function
```
params have a dash after the name

inlined commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+ * Handle HTTPS requests
+ * @param opts Options to set on this Cloud Function
+ * @param handler A function that takes a request and response object, same signature as an Express app.
+ */
```
Please add a @returns.

TheIronDev commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
 
+/**
+ * Handles HTTPS requests.
+ */
```
Please document req/res

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+  /** HTTP functions can override global options and can specify multiple regions to deploy to. */
   region?: SupportedRegion | string | Array<SupportedRegion | string>;
+  /** If true, allows CORS on requests to this function.
+   * If this is a String or ReqExp, allows requests from domains that match the provided string or RegExp.
```
Should we backtick String and RegExp (as contrasted with the lowercase "the string" below?  

Suggest "If set to `String` or . . ."  and "If set to `Array`, ..."

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+  /** If true, allows CORS on requests to this function.
+   * If this is a String or ReqExp, allows requests from domains that match the provided string or RegExp.
+   * If this is an Array, allows requests from domains matching at least one entry of the array.
+   * Defaults to true for {@link CallableFunction}s.
```
Is it obvious from the code that it defaults to "false" for HTTP triggers, or should we add that?

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+ * Handles HTTPS requests.
+ * @param opts - Options to set on this Cloud Function
+ * @param handler - A function that takes a {@link Request} and response object, same signature as an Express app.
+ * @returns A Cloud Function that you can export and deploy.
```
Suggest "A function that you can..."

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
+/**
+ * Handles HTTPS requests.
+ * @param handler - A function that takes a {@link Request} and response object, same signature as an Express app.
+ * @returns A Cloud Function that you can export and deploy.
```
"A function that you can..."

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -160,10 +189,21 @@ export function onRequest(
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param opts - Options to set on this Cloud Function.
+ * @param handler - A function that takes a {@link CallableRequest}.
+ * @returns A Cloud Function that you can export and deploy.
```
"A function that you can..."

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -160,10 +189,21 @@ export function onRequest(
 
+/**
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param opts - Options to set on this Cloud Function.
```
Suggest "this function."

egilmorez commented on src/v2/providers/https.ts:
```diff
@@ -160,10 +189,21 @@ export function onRequest(
+/**
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param handler - A function that takes a {@link CallableRequest}.
+ * @returns A Cloud Function that you can export and deploy.
```
"A function that you can..."

egilmorez reviewed (commented):
Some thoughts Joe, thanks!

Kudos for staging and eyeballing the lists (I'm assuming you did that).

joehan commented:
@egilmorez Addressed feedback. I haven't staged the list changes yet - I just manually reviewed the MD files and verified that the dcogen added linebreaks where expected.

egilmorez reviewed (approved):
LG, thanks!
